.github/workflows/merge_workflow.yml

```diff
@@ -0,0 +1,103 @@
+name: CI/CD
+
+on:
+  push:
+    branches: [main]
+    tags: ['*']
+  pull_request:
+    branches: [main]
+
+env:
+  SCALA_VERSION: 2.13.8
+  JAVA_VERSION: 11
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        test-type: ['scala', 'python']
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK
+      if: matrix.test-type == 'scala'
+      uses: actions/setup-java@v2
+      with:
+        java-version: ${{ env.JAVA_VERSION }}
+        distribution: 'adopt'
+    - name: Cache SBT dependencies
+      if: matrix.test-type == 'scala'
+      uses: actions/cache@v2
+      with:
+        path: | 
+          ~/.ivy2/cache
+          ~/.sbt
+          ~/.m2
+        key: ${{ runner.os }}-sbt-${{ hashFiles('**/build.sbt') }}
+    - name: Run Scala tests
+      if: matrix.test-type == 'scala'
+      run: |
+        sbt -Dsbt.log.noformat=true scalafmtCheckAll
+        sbt -Dsbt.log.noformat=true +test
+    - name: Set up Python
+      if: matrix.test-type == 'python'
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Cache pip dependencies
+      if: matrix.test-type == 'python'
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+    - name: Install Python dependencies
+      if: matrix.test-type == 'python'
+      run: |
+        cd python
+        make
+        pip install --only-binary=numpy,scipy -r requirements.txt
+        pip install -r test_requirements.txt
+    - name: Run Python tests
+      if: matrix.test-type == 'python'
+      run: |
+        cd python
+        nosetests
+
+  release:
+    needs: test
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/'))
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK
+      uses: actions/setup-java@v2
+      with:
+        java-version: ${{ env.JAVA_VERSION }}
+        distribution: 'adopt'
+    - name: Release to Maven Central
+      env:
+        PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
```

Comment:
what are these secrets? Have you set them? How do they get used?

Reply:
These are for deploying and they are set in Github